### PR TITLE
Extra message on module description

### LINKED ISSUE #28329: Module manager page, Allow to add content after a module description
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

module manager en savoir plus

In 1.7.8, we had a "Read more" link to display a modal with informations about the module

### Describe the solution you'd like

A hook to allow adding content after the module's description

### Alternatives you've considered

_No response_

### Additional context

_No response_

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

### PR DESCRIPTION
**Pull request by sowbiba**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | This PR adds the hook to display extra message after module description in module manager
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #28329 
| How to test?      | Register the hook actionListModules in a module and add a text/link/anything to the additional_descripton property of the modules. or you can use [prestashopexamplemodule.zip](https://github.com/PrestaShop/PrestaShop/files/8805312/prestashopexamplemodule.zip)<br>If you use the example module, you'll have in front of each module a text with "this is my additional desc"

### CODE REVIEW

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/card_list.html.twig
Comment (PululuK) [inline src/PrestaShopBundle/Resources/views/Admin/Module/Includes/card_list.html.twig]: Why hook ?
I suggest to add a new module property or method something like `getExtraDescription`

**Review by PululuK — DISMISSED** [review]

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/card_list.html.twig
Comment (sowbiba) [reply src/PrestaShopBundle/Resources/views/Admin/Module/Includes/card_list.html.twig]: this feature won't be native to Prestashop Open source anymore as the content of "what was here" comes from addons. I understand your suggestion but will it be accepted if not used in PS ?

Comment (kpodemski) [reply src/PrestaShopBundle/Resources/views/Admin/Module/Includes/card_list.html.twig]: @sowbiba, it might end up being used by other developers, too :) from what I know, a new API implementation for getting Modules allows us to connect with any API we want, not only Addons

Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Module/Includes/card_list.html.twig]: @PululuK suggestion sounds good 👍

**Review by Progi1984 — COMMENTED** [review]
Miss an autoupgrade PR :)

File: src/Adapter/Module/AdminModuleDataProvider.php
Comment (Progi1984) [inline src/Adapter/Module/AdminModuleDataProvider.php]: Please define the BC Break :)

File: classes/module/Module.php
Comment (Progi1984) [inline classes/module/Module.php]: The naming is hard. `extra` is not clear. Could you replace it by `additionalDescription` ? Thanks

**Review by Progi1984 — COMMENTED** [review]
Two comments :) Thanks

**Review by Progi1984 — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Review by khouloudbelguith — COMMENTED** [review]
Hi @sowbiba,

I used the example module with your PR but I don't have in front of each module a text with a link and in front of "presatshopexamplemodule" another text "This is the Extra Description added by the module itself".
I tried to use this module https://github.com/PrestaShop/ps_qualityassurance/
image
=> same issue
I attached a screen record

https://user-images.githubusercontent.com/16067358/168803295-5c63b6d1-fefe-4b3a-8b44-9d1776bf785c.mp4

Thanks to check and feedback.

**Comment by intraordinaire** [comment]
Hello @khouloudbelguith,
The hook must return an array of `key => value` where the `key` is the module technical name, and the `value`, the additional description.
Eg : 
```php
return [
    'statsproduct' => 'khouloud',
];
```
With this, you should be able to see "khouloud" next to the "Product details" module.

I also attach an updated version of the example module, as the hook name has changed, the module needed an update too.
[prestashopexamplemodule.zip](https://github.com/PrestaShop/PrestaShop/files/8769840/prestashopexamplemodule.zip)

File: src/Adapter/Module/AdminModuleDataProvider.php
Comment (atomiix) [inline src/Adapter/Module/AdminModuleDataProvider.php]: I'm sorry but I don't see why you're using a hook? Why not doing what @PululuK said?

**Review by atomiix — CHANGES_REQUESTED** [review]

File: src/Adapter/Module/AdminModuleDataProvider.php
Comment (PululuK) [reply src/Adapter/Module/AdminModuleDataProvider.php]: It's not the first time that my point of view has been ignored.
My opinion is irrelevant in this project I think.
I'm just a poor external maintainer.

Comment (atomiix) [reply src/Adapter/Module/AdminModuleDataProvider.php]: @PululuK there's no difference between external and internal maintainers, I already heard the same kind of comment from a PrestaShop's employee 😄

I know everyone is not comfortable with the "request changes" to block the PR in order to be sure to discuss your idea, but sometimes it's worth it 🙂.

Comment (sowbiba) [reply src/Adapter/Module/AdminModuleDataProvider.php]: :D @PululuK your POV is not ignored at all. Ok, I added a extraDescription field but my need is that MBO fills the extra description, not only the module itself. I'll be more than happy to know how a module can set an attribute to the collection of modules of the ModuleManager page without a hook or an override 

If you look well, with what I did, the module can add an extraDescription itself but another module can add extraDescription in order to be displayed in front of each module in the ModuleManager

Believe me, I'm always happy to do what others contributors/maintainers think if it's in the interest of the project. I have enough experience to know that reviews are not personal but an opinion exchange

Comment (sowbiba) [reply src/Adapter/Module/AdminModuleDataProvider.php]: Maybe you guys don't understand the need. Please read again the issue and ask for answers ;)

Comment (atomiix) [reply src/Adapter/Module/AdminModuleDataProvider.php]: What I don't understand is why you're not doing it like it's done for the module description (or even displayName) itself:
In the `class/module/Module.php` file you have the `description` property.
In the `src/Adapter/Module/Module.php` file you can have a `description` key in the `$attribute` property.
Either the module by itself has the property `description` filled, either you can fill it from an API [like we did for ps_distributionapiclient](https://github.com/PrestaShop/ps_distributionapiclient/blob/main/src/DistributionApi.php#L81).

Also, be careful, it seems [there's already an `$additional_description` property in the `src/classes/Module.php`](https://github.com/PrestaShop/PrestaShop/blob/develop/classes/module/Module.php#L95)

Comment (sowbiba) [reply src/Adapter/Module/AdminModuleDataProvider.php]: in ps_distributionapiclient, `getModulesList` is an hook action, isn't it ?

You have an entry to edit the modules' properties. In ModuleManager I don't have that kind of way to impact the displayed items, unless I add a hook

**Review by atomiix — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]
Ok for me but let's wait for @PululuK review too 👍

**Review by PululuK — APPROVED** [review]

**Comment by sowbiba** [comment]
Thanks guys :D ❤️

**Review by Progi1984 — APPROVED** [review]

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/card_list.html.twig
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/views/Admin/Module/Includes/card_list.html.twig:93]: Just a question since we use `|raw` here isn't there a risk of injecting malicious code?
@NeOMakinG what do you think?

**Review by khouloudbelguith — APPROVED** [review]
Hi @sowbiba,

Automatic tests are ok :heavy_check_mark: 
I installed the module 
[prestashopexamplemodule.zip](https://github.com/PrestaShop/PrestaShop/files/8806389/prestashopexamplemodule.zip)
I tried with an external link => ok

https://user-images.githubusercontent.com/16067358/171207545-6c0d9701-eaaf-477b-b153-9931d7e58ad1.mp4

I tried with a text => ok

https://user-images.githubusercontent.com/16067358/171207577-2766b005-c338-46e3-9292-21fa45939e73.mp4

All shops => ok :heavy_check_mark: 
image

Thanks!

**Comment by Progi1984** [comment]
Thanks @sowbiba & @khouloudbelguith

**Pull request merged**
